python/gstaichi/lang/_wrap_inspect.py

```diff
@@ -19,8 +19,10 @@
 import inspect
 import os
 import tempfile
+from typing import Callable
 
 import dill
+from pydantic import BaseModel
```

Comment:
Why using pydantic for this? For stricter enforcement? Because annotated frozen dataclass are supported natively in python.

Reply:
because I need a way to parse the json back into objects. I'm simply using what we used to use in my old company. I did check briefly using dataclasses to parse to objects, but looks like it will only parse the top level object, and none of the children? Have to recurse by hand?

Reply:
As you want. Be bit overkill for your need but it does the job just right.